tests/recipes/test_full_finetune_single_device.py

```diff
@@ -74,9 +74,6 @@ def test_loss(self, compile, config, model_type, ckpt_type, tmpdir, monkeypatch)
         ckpt_dir = ckpt_path.parent
         log_file = gen_log_file_name(tmpdir)
 
-        # To workaround https://github.com/pytorch/torchtune/issues/676
-        if compile:
-            os.environ["TORCH_COMPILE_BACKEND"] = "aot_eager"
```

Comment:
do we enforce inductor as our backend anywhere? if not, should we?

Reply:
I jsut saw that you tested it with a commit, but maybe we could enforce it here to use with inductor backend? Or add inductor to some defaults.py, and always use the default_compile_backend if defaults.py?

Reply:
> do we enforce inductor as our backend anywhere? if not, should we?

We may still wanna allow users the flexibility to change it, but in my other PR I added inductor as the default in our compile utilities [here](https://github.com/pytorch/torchtune/blob/c169bcdddaeee94fb4b599c1f083f94ee1a76143/torchtune/training/_compile.py#L38). 

> I jsut saw that you tested it with a commit, but maybe we could enforce it here to use with inductor backend? Or add inductor to some defaults.py, and always use the default_compile_backend if defaults.py? Just trying to prevent some silent error in case, for some reasons, the default of the CI machine is not inductor

I'm ambivalent about putting it in the actual test. Strictly speaking it may be safest to explicitly set it, but in order for this kind of silent error we would need (a) to remove (or change) the default in the compile utils and (b) the CI runner to have its own custom environment variables defined somehow. Based on that it's low-risk imo, but lmk if you disagree and I can add it